js/api/Surveys.js

```diff
@@ -139,7 +159,8 @@ function clearSurveyCache(exclusions) {
 
     // Standard JS array.filter doesn't work with these Realm objects, so we have to take care of this filtering manually.
     // Current version of Realm.io does not support exclusion queries for strings.
-    for (var i = surveys.length - 1; i >= 0; i--) {
+    const surveysLength = surveys.length;
+    for (var i = surveysLength - 1; i >= 0; i--) {
```

Comment:
That is a very good point. It looks like a habit I'll have to learn to break.
I'll update these next.